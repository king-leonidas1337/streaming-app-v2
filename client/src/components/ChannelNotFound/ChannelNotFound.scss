.channel-not-found {
  display: flex;
  justify-content: center;
  align-items: center;
  color: var(--color-main-dark);
  height: 100vh;
  font-size: 36px;
}

.channel-not-found span {
  font-weight: bold;
  font-style: italic;
}