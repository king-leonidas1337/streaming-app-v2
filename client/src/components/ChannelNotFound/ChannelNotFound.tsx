import React from 'react';
import './ChannelNotFound.scss';

interface ChannelNotFoundProps {
	username?: string;
}

const ChannelNotFound: React.FC<ChannelNotFoundProps> = props => {
	return (
		<div className="channel-not-found">
			Channel -&nbsp;<span>{props.username}</span>&nbsp;- does not exist ;(
		</div>
	);
};

export default ChannelNotFound;
