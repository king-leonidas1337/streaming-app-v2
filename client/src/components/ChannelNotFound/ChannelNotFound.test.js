import React from 'react';
import { render } from '@testing-library/react';
import faker from 'faker';
import '@testing-library/jest-dom/extend-expect';

import ChannelNotFound from './ChannelNotFound';

describe('ChannelNotFound component', () => {
  it('renders passed name', () => {
    const username = faker.internet.userName();

    const { getByText } = render(<ChannelNotFound username={username} />);

    expect(getByText(username)).toBeInTheDocument();
  })

  it('matches snapshot', () => {
    const { container } = render(<ChannelNotFound />);

    expect(container).toMatchSnapshot();
  })
})