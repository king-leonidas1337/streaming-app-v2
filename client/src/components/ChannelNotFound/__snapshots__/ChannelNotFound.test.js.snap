// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelNotFound component matches snapshot 1`] = `
<div>
  <div
    class="channel-not-found"
  >
    Channel - 
    <span />
     - does not exist ;(
  </div>
</div>
`;
